File: minStack.py
class MinStack(object):

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.storage = []
        self.minStorage = []
        

    def push(self, x):
        """
        :type x: int
        :rtype: void
        """
        self.storage.append(x)
        if (x < self.minStorage[len(self.minStorage) - 1]):
            self.minStorage.append(x)
        else:
            self.minStorage.append(self.minStorage[len(self.minStorage) - 1])
        

    def pop(self):
        """
        :rtype: void
        """
        self.storage.pop()
        self.minStorage.pop()
        

    def top(self):
        """
        :rtype: int
        """
        return self.storage[len(self.storage) - 1]
        

    def getMin(self):
        """
        :rtype: int
        """
        return self.minStorage[len(self.minStorage) - 1]
        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: minstack.js
var MinStack = function () {
    this.stack1 = [];
    this.stack2 = [];
};

const getLastOfArray = (arr) => {
    return arr[arr.length - 1];
}

/** 
 * @param {number} val
 * @return {void}
 */
MinStack.prototype.push = function (val) {
    this.stack1.push(val);

    if (this.stack2.length > 0) {
        this.stack2.push(Math.min(getLastOfArray(this.stack2), val));
    } else {
        this.stack2.push(val);
    }
};

/**
 * @return {void}
 */
MinStack.prototype.pop = function () {
    this.stack2.pop();
    return this.stack1.pop();
};

/**
 * @return {number}
 */
MinStack.prototype.top = function () {
    return getLastOfArray(this.stack1);
};

/**
 * @return {number}
 */
MinStack.prototype.getMin = function () {
    return getLastOfArray(this.stack2);
};

/** 
 * Your MinStack object will be instantiated and called as such:
 * var obj = new MinStack()
 * obj.push(val)
 * obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.getMin()
 */